### DISCUSSION TITLE
Running EJBCA with SSL/TLS on the front end - what's the recommendation?

### DISCUSSION THREAD

**Discussion by tfboy** [Keyfactor/ejbca-ce#347] [Q&A]
I guess this question gets asked quite a lot and I've read through many threads with many options :) However, it's still unclear to me which is the recommended route. The best route will also depend on which type of install of EJBCA - either the docker version or bare metal...

I've installed many services behind a reverse proxy (both Apache and Nginx) which acts as the SSL termination before forwarding onto the service on the same machine.

The guides I've read through discussing EJBCA suggest either [Apache as a proxy](https://doc.primekey.com/ejbca/ejbca-integration/integrating-with-third-party-applications/setting-up-an-apache-web-server-as-a-proxy) or [Apache with mod_jk](https://doc.primekey.com/ejbca/ejbca-integration/integrating-with-third-party-applications/setting-up-an-apache-web-server-with-mod_jk). However, both of these guides seem to suggest this is only with the bare metal install and not the docker version.

I've tried with the docker version, but I can't figure out how to pass through the certificate and the client cert for authentication on the admin page. The proxy guide has `bin/ejbca.sh ra addendentity apache-ssl foo123` command, but this doesn't work on my docker version - `apache-ssl` isn't a valid option.

I'd be happy to look at the mod_jk route, but not sure how to do this as Apache is running on the same machine as the docker. If Apache is doing the SSL termination on port 443 then surely I need to modify or remove the 443->8443 port mapping inside my docker-compose yaml file. Or maybe the mod_jk bypasses this completely and talks to the database but I don't see how this would work as the docker only seems to expose MySQL on port 3306...

So I guess my question boils down to is there an install guide for EJBCA on docker with HTTPS SSL/TLS termination so I can use a recognised certificate issued by EJBCA instead of the self-signed one, especially for the RA web access. Or should I abandon the docker route and do the bare metal? I'd be happy to do the latter although it seems rather involved.

**Comment by primetomas** [comment]
The container is designed for that. 
Check the documentation on DockerHub: https://hub.docker.com/r/keyfactor/ejbca-ce
Section "Using the container behind a proxy"

**Reply by tfboy** [reply]
Ah thanks, I missed that.
Is there any more info available with a more detailed explanation, or an example setting it up? I've not used AJP before and from the link you provided, there's no real indication whether it's ajp OR http or whether it's ajp AND http.
I only have a single instance using docker and would like to terminate the SSL on the same machine.

**Comment by primetomas** [comment]
You can check deployment examples on GitHub:https://github.com/Keyfactor/keyfactorcommunity
or more advanced things like the Community Helm chart: https://github.com/Keyfactor/ejbca-community-helm
